contracts/TrustToken.sol

```diff
@@ -11,21 +9,42 @@ import "./ValTokenWithHook.sol";
  * in order to prevent rewards from getting stuck in the remainder on division.
  * Tolerates dilution to slash stake and accept rewards.
  */
-contract TrustToken is ValTokenWithHook, ClaimableContract {
+contract TrustToken is ERC20 {
+
+    /**
+     * @dev initalize trusttoken and give ownership to sender
+     * This is nessesary to set ownership for proxy 
+     */
+    function initalize() public {
+        owner_ = msg.sender;
+    }
+
+    /**
+     * @dev mint TRU
+     * Can never mint more than MAX_SUPPLY = 1.45 billion
+     */
+    function mint(address _to, uint256 _amount) external onlyOwner {
+        if (_totalSupply <= MAX_SUPPLY) {
```

Comment:
@hal909 shouldn't this be _totalSupply+_amount <= MAX_SUPPLY?

Because otherwise you could exceed max amount. Also why if else and not require? Just curious, not sure that one or the other is better.